src/main/webapp/assets/modules/get-promoted/gp-form.mod.js

```diff
@@ -1,28 +1,34 @@
-techlooper
-  .directive("getPromotedForm", function ($http, vnwConfigService) {
-    return {
-      restrict: "E",
-      replace: true,
-      templateUrl: "modules/get-promoted/gp-form.tem.html",
-      link: function (scope, element, attr, ctrl) {
-        scope.doPromotion = function () {
-          if (!scope.promotionForm.$valid) {
-            return false;
-          }
-          scope.masterPromotion = angular.copy(scope.promotionInfo);
-          $http.post("getPromoted", scope.promotionInfo)
-            .success(function (data, status, headers, config) {
-              scope.masterPromotion.result = data;
-              scope.changeState('result');
-            });
-        }
+techlooper.directive("getPromotedForm", function ($http, vnwConfigService, $location) {
+  return {
+    restrict: "E",
+    replace: true,
+    templateUrl: "modules/get-promoted/gp-form.tem.html",
+    link: function (scope, element, attr, ctrl) {
+      var campaign = $location.search();
 
-        scope.$watch("masterPromotion", function (newVal, oldVal) {
-          if (!newVal && !oldVal) return false;
-          scope.masterPromotion.jobLevelIds = vnwConfigService.getJobLevelIds(scope.masterPromotion.jobLevelId);
-          scope.masterPromotion.jobLevelTitle = vnwConfigService.getJobLevelText(scope.masterPromotion.jobLevelId);
-          scope.masterPromotion.jobCategoryTitle = vnwConfigService.getIndustryTexts(scope.masterPromotion.jobCategoryIds).join(" | ");
-        });
+      scope.doPromotion = function () {
+        if (!scope.promotionForm.$valid) {
+          return false;
+        }
+        scope.masterPromotion = angular.copy(scope.promotionInfo);
+        $http.post("getPromoted", scope.promotionInfo)
+          .success(function (data, status, headers, config) {
```

Comment:
'config' is defined but never used.<br>'headers' is defined but never used.<br>'status' is defined but never used.